RedHatInsights/rhsm-subscriptions
# SWATCH-497 Update hbi queries to use org

https://issues.redhat.com/browse/SWATCH-497

**NOTE:**
* This PR requires that org_id be set on the AccountConfig (account_config table via opt-in). If the org_id is null, an exception is thrown
* The following PR should be merged and verified working before this is merged (https://github.com/RedHatInsights/rhsm-subscriptions/pull/1353)

**How To Test**

Deploy dependant services:
```
$ podman-compose up -d
```

Import some host data into the local HBI database. 
**The CSV reference here can be found in the comments of SWATCH-497.**
```
$ cat 6089719-hbi-hosts.csv | psql -h 0.0.0.0 -U insights insights -c "copy hosts(id, account, display_name, created_on, modified_on, facts, tags, canonical_facts, system_profile_facts, ansible_host, stale_timestamp, reporter, per_reporter_staleness, org_id) from STDIN delimiter E'\t' csv header;"
```

Run the application:
```
./gradlew clean :bootRun
```

Opt-in the account/org for the test data imported above.
```
curl 'http://localhost:9000/hawtio/jolokia/' -H 'Content-Type: application/json' -d '{"type":"exec","mbean":"org.candlepin.subscriptions.jmx:name=optInJmxBean,type=OptInJmxBean","operation":"createOrUpdateOptInConfig(java.lang.String,java.lang.String,boolean,boolean,boolean)","arguments":["6089719", "11789772", true, true, true]}'
```

Run a nightly tally for all opted in accounts:
```
curl 'http://localhost:9000/hawtio/jolokia/' -H 'Content-Type: application/json' -d '{"type":"exec","mbean":"org.candlepin.subscriptions.jmx:name=tallyJmxBean,type=TallyJmxBean","operation":"tallyConfiguredAccounts()","arguments":[]}'
{"request":{"mbean":"org.candlepin.subscriptions.jmx:name=tallyJmxBean,type=TallyJmxBean","type":"exec","operation":"tallyConfiguredAccounts()"},"value":null,"timestamp":1661526268,"status":200}[mstead@redcanoe ~]$ curl 'http://localhost:9000/hawtio/jolokia/' -H 'Content-Type: application/json' -d '{"type":"exec","mbean":"org.candlepin.subscriptions.jmx:name=tallyJmxBean,type=TallyJmxBean","operation":"tallyConfiguredAccounts()","arguments":[]}'
```

Check the DB to ensure that hosts were pulled from HBI and snapshots were created:
```
psql -h 0.0.0.0 -U rhsm-subscriptions rhsm-subscriptions -c "select count(*) from tally_snapshots where account_number='6089719';"

psql -h 0.0.0.0 -U rhsm-subscriptions rhsm-subscriptions -c "select h.account_number, h.org_id, m.instance_id, uom, value from hosts h, instance_measurements m where h.id=m.instance_id and account_number='6089719' and org_id='11789772'"
```
